pyqg/kernel.pyx

```diff
@@ -347,6 +388,53 @@ cdef class PseudoSpectralKernel:
                                     self._ikQy[k,i] * self.ph[k,j,i] )
         return
 
+    def _do_uv_subgrid_parameterization(self):
+        self.__do_uv_subgrid_parameterization()
+
+    cdef __do_uv_subgrid_parameterization(self):
+        """Add the uv subgrid parameterization"""
+        cdef Py_ssize_t k, j, i
+        du, dv = self.uv_parameterization(self)
+        cdef DTYPE_real_t [:, :, :] du_view = du
+        cdef DTYPE_real_t [:, :, :] dv_view = dv
+        # convert to cython memory view
+        self.du[:, :, :] = du_view
+        self.dv[:, :, :] = dv_view
+        # convert to spectral space
+        self.fft_du_to_duh()
+        self.fft_dv_to_dvh()
```

Comment:
Could we not just use function-local variables here?

Reply:
See comment above -- I don't have a great sense of the C performance considerations here.